tests/unit/test_settings_updates.py

```
@@ -345,3 +346,90 @@ def start(self):
+        mod._set_update_state(False, None)
+        monkeypatch.setattr(mod, "_systemd_available", lambda: False)
+        monkeypatch.setattr(mod, "_get_update_script_path", lambda: None)
+        monkeypatch.setattr(
+            mod, "_start_update_fallback_thread", lambda sp, target_tag=None: None
+        )
+
+        lock_held_when_running_flipped: list[bool] = []
+        real_dict = mod._UPDATE_STATE
+
+        class _SpyDict(dict):
+            def __setitem__(self, key, value):
```

Ruff (0.15.9)

[warning] Missing return type annotation for special method `__setitem__`

Add return type annotation: `None`

(ANN204)